lib/rubocop/path_util.rb

```diff
@@ -15,7 +15,11 @@ def relative_path(path, base_dir = PathUtil.pwd)
       end
 
       path_name = Pathname.new(File.expand_path(path))
-      path_name.relative_path_from(Pathname.new(base_dir)).to_s
+      begin
+        path_name.relative_path_from(Pathname.new(base_dir)).to_s
+      rescue ArgumentError
```

Comment:
It rescues unexpected errors. For example, `Pathname#relative_path_from` raises an error when relative path and absolute path are mixed. It will conceals RuboCop's bug.

```ruby
require 'pathname'

Pathname('foo/').relative_path_from(Pathname('/bar'))
```

```bash
$ ruby test.rb
/home/pocke/.rbenv/versions/trunk/lib/ruby/2.6.0/pathname.rb:522:in `relative_path_from': different prefix: "" and "/bar" (ArgumentError)
	from test.rb:3:in `<main>'
```

So, I think it should check start of the path with regexp or something expressly.

Reply:
File.expand_path converts the path to absolute form

Reply:
True, but it is one of the examples that raises ArgumentError. ArgumentErorr is raised by many reasons other than the example, and the reasons are bugs in almost cases. So I think it should not rescue ArgumentError.

Reply:
I'm not sure I understand.
From a quick look at the code, `ArgumentError` is raised in exactly 2 cases:
1. Different prefix (which is what this patch covers). Given that both paths are absolute, this should only be triggered on Windows with different drives.
2. If the base directory (the argument to `relative_path_from`) contains `..`, but `File.expand_path` removes all these as well, so this shouldn't happen.
Can you demonstrate a real bad case?

Reply:
Currently, probably it raises `ArgumentError` only if the paths on different drives. But I think rescuing has big maintenance cost.
If we use `rescue ArgumentError`, we should be careful that `ArgumentError` is not raised for other reasons in the feature. For example, we cannot notice that we mistake arity.
So I think the rescuing has big maintenance cost. We can reduce the cost if it checks drive name with regular expressions.

I guess we can implement the regexp easily, but I'm not sure the difficulty because I don't use Windows. So if it's difficult, please tell me the problem. maybe we should use rescuing if implementaion of the regexp checking is hard.

Reply:
It is difficult, because Pathname also detects other things like UNC names (`\\server\share` against `\\otherserver\othershare`). I doubt you want to duplicate all this logic from Pathname, when it provides you with an exception just for that.